content/en/docs/concepts/storage/volume-snapshots.md

```diff
@@ -141,13 +142,44 @@ spec:
   driver: hostpath.csi.k8s.io
   source:
     snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
+  sourceVolumeMode: Filesystem
   volumeSnapshotRef:
     name: new-snapshot-test
     namespace: default
 ```
 
 `snapshotHandle` is the unique identifier of the volume snapshot created on the storage backend. This field is required for the pre-provisioned snapshots. It specifies the CSI snapshot id on the storage system that this `VolumeSnapshotContent` represents.
 
+`sourceVolumeMode` is the mode of the volume whose snapshot is taken. The value of the `sourceVolumeMode` field can be either `Filesystem` or `Block`. If the source volume mode is not specified, Kubernetes treats the snapshot as if the source volume's mode is unknown. Support for this field can only be enabled in VolumeSnapshot client v6.0.0 and higher. Visit the [VolumeSnapshot Release page](https://github.com/kubernetes-csi/external-snapshotter/releases) for more information.
```

Comment:
Do you need to be using Kubernetes v1.24 to set this field?

Reply:
`VolumeSnapshot`s 6.0 release will only be supported with a minimum k8s version of 1.24. @xing-yang can correct me if I'm wrong here.

Reply:
Do we mean:
> you need to be using `external-snapshotter` release 6.0 or later (which then requires Kubernetes v1.24)

?

Reply:
VolumeSnapshot APIs are CRDs. So this field is not really in k/k code base.  
The only thing that is in k/k code base for the snapshot feature is the DataSource field in PVC which moved to GA in 1.20 release.
But we do update K8s dependencies in https://github.com/kubernetes-csi/external-snapshotter code and release it shortly after every K8s release. VolumeSnapshot client will be released shortly after K8s 1.24 release.

Reply:
What we're trying to understand here is if there is a dependency for the version of `VolumeSnapshot`client on the Kubernetes release . You've mentioned that only the Datasource field (which moved to GA) is there in the k/k codebase. Are the other K8s dependencies that you mentioned related to other components within a release even though they're out of tree?

Reply:
VolumeSnapshot client uses a few K8s components.  See here:
https://github.com/kubernetes-csi/external-snapshotter/blob/v6.0.0-rc4/client/go.mod#L6-L9
Currently it uses 1.23 components. After K8s 1.24 is released, the dependencies will be updated to 1.24 and we'll cut the official v6.0.0 client release.